backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/ModularContractService.java

```diff
@@ -107,31 +113,31 @@ private static <T> boolean isHandlerApplicableForSettings(@NonNull final IModula
 				.anyMatch(config -> config.isMatchingClassName(handlerClassName));
 	}
 
-	private <T> void invokeWithModel(@NonNull final IModularContractTypeHandler<T> handler, final @NonNull T model, final @NonNull ModelAction action, @NonNull final FlatrateTermId flatrateTermId)
+	private <T> void invokeWithModel(
+			@NonNull final IModularContractTypeHandler<T> handler,
+			@NonNull final T model,
+			@NonNull final ModelAction action,
+			@NonNull final FlatrateTermId flatrateTermId,
+			@NonNull final LogEntryContractType logEntryContractType)
 	{
 		handler.validateDocAction(model, action);
 
-		createLogEntries(handler, model, action, flatrateTermId);
+		processLogsEnqueuer.enqueueAfterCommit(TableRecordReference.of(model), action, logEntryContractType);
 
 		handleAction(handler, model, action, flatrateTermId);
 	}
 
-	private <T> void createLogEntries(@NonNull final IModularContractTypeHandler<T> handler, final @NonNull T model, final @NonNull ModelAction action, @NonNull final FlatrateTermId flatrateTermId)
-	{
-		switch (action)
-		{
-			case COMPLETED -> handler.createLogEntryCreateRequest(model, flatrateTermId)
-					.ifPresent(contractLogDAO::create);
-			case VOIDED, REACTIVATED, REVERSED -> handler.createLogEntryReverseRequest(model, flatrateTermId)
-					.ifPresent(contractLogDAO::reverse);
-		}
-	}
-
-	private <T> void handleAction(@NonNull final IModularContractTypeHandler<T> handler, final @NonNull T model, final @NonNull ModelAction action, @NonNull final FlatrateTermId flatrateTermId)
+	private <T> void handleAction(
+			@NonNull final IModularContractTypeHandler<T> handler,
+			@NonNull final T model,
+			@NonNull final ModelAction action,
+			@NonNull final FlatrateTermId flatrateTermId)
 	{
-		switch (action)
+		switch (action) //FIXME
```

Comment:
fix it or remove it :)

Reply:
don't have time for it now, but I'll refactor with a following issue